### PR TITLE
Initial memory counter for unary client.

### PR DESCRIPTION
**Pull request by nicolasnoble**
There's a couple of memory leaks server-side however:

```
==12456== 40 bytes in 1 blocks are possibly lost in loss record 1 of 2
==12456==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==12456==    by 0x403BED: guard_malloc (memory_counters.c:59)
==12456==    by 0x403C35: guard_realloc (memory_counters.c:67)
==12456==    by 0x4135D0: gpr_realloc (alloc.c:75)
==12456==    by 0x4AEC9C: add_epoll_fd_to_global_list (ev_poll_and_epoll_posix.c:1490)
==12456==    by 0x4AF5A3: epoll_become_multipoller (ev_poll_and_epoll_posix.c:1718)
==12456==    by 0x4AE624: basic_do_promote (ev_poll_and_epoll_posix.c:1080)
==12456==    by 0x41E49B: grpc_exec_ctx_flush (exec_ctx.c:54)
==12456==    by 0x4AE09E: pollset_work (ev_poll_and_epoll_posix.c:942)
==12456==    by 0x41E233: grpc_pollset_work (ev_posix.c:108)
==12456==    by 0x40A0BA: grpc_completion_queue_next (completion_queue.c:342)
==12456==    by 0x4039A3: main (memory_server.c:227)
==12456==
==12456== 520 bytes in 1 blocks are definitely lost in loss record 2 of 2
==12456==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==12456==    by 0x403BED: guard_malloc (memory_counters.c:59)
==12456==    by 0x403C35: guard_realloc (memory_counters.c:67)
==12456==    by 0x4135D0: gpr_realloc (alloc.c:75)
==12456==    by 0x406429: publish_app_metadata (call.c:859)
==12456==    by 0x406588: recv_initial_filter (call.c:885)
==12456==    by 0x4119B1: grpc_metadata_batch_filter (metadata_batch.c:154)
==12456==    by 0x406E04: receiving_initial_metadata_ready (call.c:1082)
==12456==    by 0x40CD8C: server_on_recv_initial_metadata (server.c:718)
==12456==    by 0x41C70B: hs_on_recv (http_server_filter.c:175)
==12456==    by 0x41E49B: grpc_exec_ctx_flush (exec_ctx.c:54)
==12456==    by 0x4AE09E: pollset_work (ev_poll_and_epoll_posix.c:942)
```

### CODE REVIEW

File: test/core/memory/memory_client.c [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    event = grpc_completion_queue_next(cq, gpr_inf_future(GPR_CLOCK_REALTIME),
+                                       NULL);
+  } while (event.type != GRPC_QUEUE_SHUTDOWN);
+  memory_probe("grpc_completion_queue_next*n");
+  grpc_completion_queue_destroy(cq);
+  memory_probe("grpc_completion_queue_destroy");
+  grpc_byte_buffer_destroy(the_buffer);
+  memory_probe("grpc_byte_memory_buffer_destroy");
+  gpr_slice_unref(slice);
+
```
Comment (ctiller) [inline test/core/memory/memory_client.c]: grpc_metadata_array_destroy(&initial_metadata);
grpc_metadata_array_destroy(&trailing_metadata);

Comment (nicolasnoble) [inline test/core/memory/memory_client.c]: I think you meant that in memory_server.c ? :)

Comment (nicolasnoble) [inline test/core/memory/memory_client.c]: Actually, no, these are missing too.

**Comment by hcaseyal** [comment]
@nicolasnoble May I close this PR?

**Pull request closed without merging**